File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        
        for (int i = 0; i < numbers.length; i++) {
            if (numbers[i] % 2 == 0) {
                answer[i] = numbers[i] + 1;
            } else {
                StringBuilder sb = new StringBuilder(Long.toBinaryString(numbers[i]));
                int zero = sb.lastIndexOf("0");
                if (zero == -1) {
                    sb.setCharAt(0, '0');
                    sb.insert(0, "1");
                } else {
                    sb.setCharAt(zero, '1');
                    sb.setCharAt(zero + 1, '0');
                }
                answer[i] = Long.parseLong(sb.toString(), 2);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers: list) -> list:
    answer = []
    
    for number in numbers:
        if number % 2 == 0:
            answer.append(number + 1)
        else:
            binary = list(bin(number))
            binary[1] = '0'
            for i in range(len(binary) - 1, 0, -1):
                if binary[i] == '0':
                    binary[i] = '1'
                    binary[i + 1] = '0'
                    break
            answer.append(int(''.join(binary), 2))
    
    return answer